pkg/dns/vips_allocator.go

```diff
@@ -104,17 +107,19 @@ func (d *VIPsAllocator) createOrUpdateVIPConfigs(meshes ...string) (errs error)
 			return err
 		}
 
-		updated, updError := UpdateMeshedVIPs(global, meshed, ipam, serviceSet)
-		if !updated {
-			return updError
+		changed, err := UpdateMeshedVIPs(global, meshed, ipam, serviceSet)
+		if err != nil {
+			// Error might occur only if we run out of VIPs. There is no point to pass it through,
+			// we must notify user in logs and proceed
+			vipsAllocatorLog.Error(err, "failed to allocate new VIPs")
```

Comment:
shouldn't we return after this one?

Reply:
No, `UpdateMeshedVIPs` could allocate some IPs and we should update them in the configmap